### PR TITLE
Added font option and optimised the client!

### PR DESCRIPTION
**Pull request by Hackcraft**
I thought this would be a pretty cool addition to 3D2D textscreens!

### CODE REVIEW

**Comment by Cherry** [comment]
A lot of cool changes here, thanks! I'll take a look as soon as I can.

File: lua/entities/sammyservers_textscreen/cl_init.lua [hunk 1]
```diff
@@ -1,64 +1,58 @@
...
 	self.lines = self.lines or {}
+//	screenInfo[self] = screenInfo[self] or {}
 	net.Start("textscreens_download")
 	net.WriteEntity(self)
 	net.SendToServer()
 end
 
 function ENT:Draw()
-	if self:GetPos():Distance(LocalPlayer():GetPos()) < render_range:GetInt() then
+	if self:GetPos():DistToSqr(LocalPlayer():GetPos()) < render_range and screenInfo[self] != nil then
```
Comment (Dellkan) [inline lua/entities/sammyservers_textscreen/cl_init.lua]: Nice catch. Should probably throw in a comment on why the render_range is multiplied earlier (it's easy to gloss over the change to DistToSqr, and to not know that it is more efficient than Distance)

File: lua/entities/sammyservers_textscreen/cl_init.lua [hunk 2]
```diff
@@ -1,64 +1,58 @@
...
+		self:DrawModel()
```
Comment (Dellkan) [inline lua/entities/sammyservers_textscreen/cl_init.lua]: Do we want the placeholder model to be drawn?

File: lua/entities/sammyservers_textscreen/cl_init.lua [hunk 3]
```diff
@@ -73,10 +67,48 @@ end
...
+		end
+
+		for i=1, #t do
+			t2[i].pos = -(totheight / 2) + curheight
+			curheight = curheight + t2[i].height
+		end
+
+		t2.tableSize = #t
+
+//		PrintTable(t2)
```
Comment (Dellkan) [inline lua/entities/sammyservers_textscreen/cl_init.lua]: By the time you're submitting a pull request, you should try to remove all commented code.

If left in, they tend to crop up over the years, while nobody remembers why it's there, who put it there, or if it's safe to remove.

**Comment by Hackcraft** [comment]
Thanks, I'm still getting use to Github myself but thanks for the feedback.

**Comment by Hackcraft** [comment]
Is there anything else I need to do so that it can be merged to the main branch?
Thanks.

**Comment by Cherry** [comment]
I still haven't had the chance to test this myself unfortunately.

I do have one question though - how does this working with the perma-saving/spawning of textscreens? I believe there should be some changes at https://github.com/Cherry/3D2D-TextScreens/blob/master/lua/autorun/textscreens_util.lua#L45-L75?

**Comment by Hackcraft** [comment]
Thanks for the hint. I've added the extra code now. Totally forgot about permanent screens :(

**Comment by Hackcraft** [comment]
Will you be able to test soon?
I can give you the IP to a server running it if that helps.
Thanks.

**Comment by Cherry** [comment]
Sorry for the delay. I'm working on reviewing it now, and will have it pushed by the end of the day if there's no issues. Thanks again!

**Comment by Cherry** [comment]
This has been merged and released. Thanks!

**Comment by Cherry** [comment]
@Hackcraft There are currently multiple reports of this change causing errors on the client. http://steamcommunity.com/sharedfiles/filedetails/?id=109643223

The specific error is:
```[ERROR] lua/entities/sammyservers_textscreen/cl_init.lua:108: 'Coolvetica outlined0' isn't a valid font

1. SetFont - [C]:-1
2. AddDrawingInfo - lua/entities/sammyservers_textscreen/cl_init.lua:108
3. func - lua/entities/sammyservers_textscreen/cl_init.lua:134
4. unknown - lua/includes/extensions/net.lua:32```

Can you investigate please?

**Pull request merged**